services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx

```diff
@@ -112,23 +112,27 @@ export const EnvironmentListItem: React.FC<EnvironmentListItemProps> = ({
             },
         });
     }, [app, env.name]);
-    const deploy = useCallback(() => {
-        if (release.version) {
-            addAction({
-                action: {
-                    $case: 'deploy',
-                    deploy: {
-                        environment: env.name,
-                        application: app,
-                        version: release.version,
-                        ignoreAllLocks: false,
-                        lockBehavior: LockBehavior.Ignore,
-                    },
+    const onConfirm = useCallback((): void => {
+        addAction({
+            action: {
+                $case: 'deploy',
```

Comment:
Since the function now is onConfirm, it makes more sense to me that the case is passed as a parameter. Either that or maybe rename the function to `onDeployConfirm`

Reply:
I combined and renamed the functions